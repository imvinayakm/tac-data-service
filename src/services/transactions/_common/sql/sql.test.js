const pg = require('knex')({ client: 'pg' });

const createSql = require('.');

// sample query
const query = pg('some_table').select('*');
const filterValues = {
  sender: 'sender',
  timeStart: 'timeStart',
  timeEnd: 'timeEnd',
  sort: 'desc',
  limit: 1,
};

const sql = createSql({ query });

describe('Sql builder', () => {
  describe('search', () => {
    it('covers case with all filters (without after)', () => {
      expect(sql.search(filterValues)).toMatchSnapshot();
    });
    it('covers case with all filters with after', () => {
      expect(
        sql.search({
          ...filterValues,
          after: {
            timestamp: 'timestamp',
            id: 'id',
            sortDirection: 'sortDirection',
          },
        })
      ).toMatchSnapshot();
    });
  });
  describe(' get', () => {
    it('works', () => {
      expect(sql.get('id')).toMatchSnapshot();
    });
  });
  describe('mget', () => {
    it('works', () => {
      expect(sql.mget(['id1', 'id2'])).toMatchSnapshot();
    });
  });
});
