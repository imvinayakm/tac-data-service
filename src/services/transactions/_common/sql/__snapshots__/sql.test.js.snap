// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sql builder  get works 1`] = `"select * from \\"some_table\\" where \\"id\\" = 'id'"`;

exports[`Sql builder mget works 1`] = `"select * from \\"some_table\\" where \\"id\\" in ('id1', 'id2')"`;

exports[`Sql builder search covers case with all filters (without after) 1`] = `"select * from \\"some_table\\" where \\"sender\\" = 'sender' and \\"time_stamp\\" >= 'timeStart' and \\"time_stamp\\" <= 'timeEnd' order by \\"time_stamp\\" desc, \\"id\\" desc limit 1"`;

exports[`Sql builder search covers case with all filters with after 1`] = `"select * from \\"some_table\\" where \\"sender\\" = 'sender' and \\"time_stamp\\" >= 'timeStart' and \\"time_stamp\\" <= 'timeEnd' and (time_stamp, id) > ('timestamp', 'id') order by \\"time_stamp\\" desc, \\"id\\" desc limit 1"`;
